src/jit/codegenxarch.cpp

```diff
@@ -7366,6 +7366,66 @@ void CodeGen::genSSE2BitwiseOp(GenTreePtr treeNode)
     inst_RV_RV(ins, targetReg, operandReg, targetType);
 }
 
+//-----------------------------------------------------------------------------------------
+// genSSE41RoundOp - generate SSE41 code for the given tree as a round operation
+//
+// Arguments:
+//    treeNode  - tree node
+//
+// Return value:
+//    None
+//
+// Assumptions:
+//     i) SSE4.1 is supported by the underlying hardware
+//    ii) tree oper is a GT_INTRINSIC
+//   iii) tree type is floating point type
+//    iv) tree oper is CORINFO_INTRINSIC_Round, _Ceiling, or _Floor
+//     v) caller of this routine needs to call genProduceReg()
+void CodeGen::genSSE41RoundOp(GenTree* treeNode)
+{
+    // i) SSE4.1 is supported by the underlying hardware
+    assert(compiler->compSupports(InstructionSet_SSE41));
+
+    // ii) tree oper is a GT_INTRINSIC
+    assert(treeNode->OperGet() == GT_INTRINSIC);
+
+    GenTree*    srcNode  = treeNode->gtGetOp1();
+
+    instruction ins      = ins_FloatRound(treeNode->TypeGet());
+    emitAttr    size     = emitTypeSize(treeNode);
+
+    regNumber   dstReg   = treeNode->gtRegNum;
+    regNumber   srcReg   = srcNode->gtRegNum;
+
+    // iii) tree type is floating point type
+    assert(varTypeIsFloating(srcNode));
```

Comment:
These checks should be at the start of the method

Reply:
I have it here so that they `srcNode` variable, which is used elsewhere, can be reused.

The asserts can either:
* all be at the beginning, but duplicate some code (it would be `assert(varTypeIsFloating(treeNode->gtGetOp1()));`)
-or-
* just after the variables are initialized

Reply:
Yeah, I guess the actual problem is that `ins` is computed too early.

Reply:
Moved `ins` and other variables after the assert.